device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -138,10 +140,8 @@ else if (this.config.getAuthenticationType() == DeviceClientConfig.AuthType.X509
                     this.iotHubUserPassword = null;
                 }
 
-                //URLEncoder follows HTML spec for encoding urls, which includes substituting space characters with '+'
-                // We want "%20" for spaces, not '+', however, so replace them manually after utf-8 encoding
                 String userAgentString = this.config.getProductInfo().getUserAgentString();
-                String clientUserAgentIdentifier = "DeviceClientType=" + URLEncoder.encode(userAgentString, "UTF-8").replaceAll("\\+", "%20");
+                String clientUserAgentIdentifier = "DeviceClientType=" + URIUtil.encodePath(userAgentString, StandardCharsets.UTF_8.name());
```

Comment:
URIUtil encodePath handles spaces correctly for our purposes, whereas URLEncoder did not, hence the switch